Common/src/main/java/com/blamejared/crafttweaker/api/ScriptLoadingOptions.java

```diff
@@ -111,8 +120,51 @@ public ScriptLoadingOptions setSource(ScriptLoadSource source) {
         return this;
     }
     
-    public static record ScriptLoadSource(ResourceLocation id) {
+    public ScriptLoadingOptions setRunConfiguration(final ScriptRunConfiguration runConfiguration) {
+        
+        this.runConfiguration = LegacyRunConfig.adapt(runConfiguration);
+        return this;
+    }
+    
+    @Deprecated(forRemoval = true) // Replace with equivalent of LegacyRunConfig#adapt
+    public void configureRun(final IgnorePrefixCasingBracketParser bep, final ScriptingEngine engine) throws CompileException {
+        
+        this.runConfiguration.configureRun(bep, engine);
+    }
+    
+    @Deprecated(forRemoval = true)
+    private void legacyConfig(final IgnorePrefixCasingBracketParser bep, final ScriptingEngine engine) throws CompileException {
+        
+        CraftTweakerAPI.LOGGER.info("The loader '{}' is using the legacy script run configuration: this will not survive a 1.19 update", this.loaderName);
+        Services.EVENT.fireRegisterBEPEvent(bep); // Backwards compatibility event, should not be used anymore
+        LegacyRunConfig.adapt(CraftTweakerDefaultScriptRunConfiguration.DEFAULT_CONFIGURATION)
+                .configureRun(bep, engine);
+    }
+    
+    public record ScriptLoadSource(ResourceLocation id) {}
     
+    @FunctionalInterface
+    public interface ScriptRunConfiguration {
+        
+        void configureRun(
+                final BiConsumer<String, BracketExpressionParser> registrationFunction,
+                final Consumer<JavaNativeModule> addingFunction,
```

Comment:
Names: 
- registrationFunction: register what where?
- addingFunction: adding What to what?